athena-opensearch/src/main/java/com/amazonaws/athena/connectors/opensearch/OpensearchMetadataHandler.java

```diff
@@ -0,0 +1,426 @@
+                }
+                found = true;
+            }
+
+            if (!found) {
+                throw new RuntimeException("Could not find table in " + tableName.getSchemaName());
+            }
+
+            // add partition columns
+            partitionSchema.getFields().forEach(schemaBuilder::addField);
+
+            return schemaBuilder.build();
+        }
+    }
+
+    private int decodeContinuationToken(GetSplitsRequest request)
+    {
+        if (request.hasContinuationToken()) {
+            return Integer.valueOf(request.getContinuationToken());
+        }
+
+        //No continuation token present
+        return 0;
+    }
+
+    private String encodeContinuationToken(int partition)
+    {
+        return String.valueOf(partition);
+    }
+
+    @Override
+    public ListTablesResponse doListTables(final BlockAllocator blockAllocator, final ListTablesRequest listTablesRequest)
+            throws Exception
+    {
+        try (Connection connection = jdbcConnectionFactory.getConnection(getCredentialProvider())) {
+            LOGGER.info("{}: List table names for Catalog {}, Table {}", listTablesRequest.getQueryId(), listTablesRequest.getCatalogName(), listTablesRequest.getSchemaName());
+            return new ListTablesResponse(listTablesRequest.getCatalogName(),
+                    listTables(connection, listTablesRequest.getSchemaName()), null);
+        }
+    }
+
+    private List<TableName> listTables(final Connection jdbcConnection, final String databaseName)
+            throws SQLException
+    {
+        try (ResultSet resultSet = getTables(jdbcConnection, databaseName)) {
+            ImmutableList.Builder<TableName> list = ImmutableList.builder();
+            while (resultSet.next()) {
+                // checks if table name is internal table ex: .kibana
+                String tableName = resultSet.getString("TABLE_NAME");
+                if (tableName.startsWith(".")) {
+                    if (!tableName.matches(DATA_STREAM_REGEX)) {
+                        continue;
+                    }
+                }
+                // update schema from null because Opensearch doesn't have schema but change later
+                // resultSet.updateString("TABLE_SCHEM", databaseName);
+                
+                // Temporary in schemaName 
+                list.add(getSchemaTableName(resultSet, databaseName));
+            }
+            return list.build();
+        }
+    }
+
+    private TableName getSchemaTableName(final ResultSet resultSet, String schemaName)
+            throws SQLException
+    {
+        String tableName = resultSet.getString("TABLE_NAME");
+        String tableSchema = resultSet.getString("TABLE_SCHEM");
+        Pattern pattern = Pattern.compile(DATA_STREAM_REGEX);
+        Matcher matcher = pattern.matcher(tableName);
+        if (matcher.find()) {
+            tableName = matcher.group(2);
+        }
```

Comment:
can you explain this? I thought you mentioned data streams are not yet supported. Also, let's avoid magic numbers or add an example in the comment for what this looks like to explain why we match on group 2